src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,131 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private int size = 0;
+    private int maxSize = 0;
+    private Object[] elementData;
+
     @Override
     public void add(T value) {
-
+        if (this.isEmpty()) {
+            elementData = new Object[DEFAULT_SIZE];
+            maxSize = DEFAULT_SIZE;
+        } else {
+            if (this.size == maxSize) {
+                maxSize = maxSize + maxSize / 2;
+                Object[] oldElementData = elementData;
+                elementData = new Object[maxSize];
+                System.arraycopy(oldElementData, 0, elementData, 0, size);
```

Comment:
```suggestion
            if (this.size == maxSize) {
                maxSize = maxSize + maxSize / 2;
                Object[] oldElementData = elementData;
                elementData = new Object[maxSize];
                System.arraycopy(oldElementData, 0, elementData, 0, size);
```

move this logic to the separate method, like resizeIfNeeded() or something else